src/esp/gfx/PbrShader.h

```diff
@@ -185,28 +190,91 @@ class PbrShader : public Magnum::GL::AbstractShaderProgram {
      * will use the node's semantic ID
      */
     InstancedObjectId = (1 << 12) | ObjectId,
+
+    /**
+     * Has ClearCoat layer.
+     */
+    ClearCoatLayer = 1 << 13,
+    // TODO : possibly break out each layer's flags to a separate flags state
+    // machine?
+    /**
+     * Has ClearCoat Texture in ClearCoat layer
+     */
+    CCLayer_CCTexture = (1 << 14) | ClearCoatLayer,
+    /**
+     * Has Roughness Texture in ClearCoat layer
+     */
+    CCLayer_RoughnessTexture = (1 << 15) | ClearCoatLayer,
+    /**
+     * Has Normal Texture in ClearCoat layer
+     */
+    CCLayer_NormalTexture = (1 << 16) | ClearCoatLayer,
+
+    CCLayer_NormalTextureScale = (1 << 17) | ClearCoatLayer,
```

Comment:
The `NormalTextureScale` flag is useless, that feature is so cheap that it makes sense to have it enabled implicitly together with `NormalTexture` (I do exactly that in the Phong shader). Thus same here for the clear coat layer.